crates/bevy_app/src/plugin_group.rs

```diff
@@ -2,6 +2,123 @@ use crate::{App, AppError, Plugin};
+/// # } }
+/// # #[derive(Default)] pub struct TickratePlugin;
+/// # impl Plugin for TickratePlugin { fn build(&self, _: &mut App) {} }
+/// # mod features {
+/// #   use bevy_app::*;
+/// #   #[derive(Default)] pub struct ForcePlugin;
+/// #   impl Plugin for ForcePlugin { fn build(&self, _: &mut App) {} }
+/// # }
+/// # mod web {
+/// #   use bevy_app::*;
+/// #   #[derive(Default)] pub struct WebCompatibilityPlugin;
+/// #   impl Plugin for WebCompatibilityPlugin { fn build(&self, _: &mut App) {} }
+/// # }
+/// # mod internal {
+/// #   use bevy_app::*;
+/// #   #[derive(Default)] pub struct InternalPlugin;
+/// #   impl Plugin for InternalPlugin { fn build(&self, _: &mut App) {} }
+/// # }
+/// plugin_group!(
+///     PhysicsPlugins {
+///         // Due to local ambiguity issues, you have to
+///         // use a colon before the plugin's name.
+///         :TickratePlugin,
+///         // Due to local ambiguity issues, you have to
+///         // use 3 colons for the last part of the path.
+///         collision::capsule:::CapsuleCollisionPlugin,
+///         velocity:::VelocityPlugin,
+///         // Add a documented feature restriction.
+///         #[cfg(feature = "external_forces")]
+///         features:::ForcePlugin,
+///         // You can add any attribute you want like this, but it won't be documented.
+///         #[custom(cfg(target_arch = "wasm32"))]
+///         web:::WebCompatibilityPlugin,
+///         // You can hide plugins from documentation.
+///         // NOTE: Due to macro limitations, you can only use
+///         //       #[doc(hidden)] for plugins at the bottom of the macro.
+///         #[doc(hidden)]
+///         internal:::InternalPlugin
+///     }
+/// );
+/// ```
+#[macro_export]
+macro_rules! plugin_group {
+    (
+        $(#[$group_meta:meta])*
+        $group:ident {
+            $(
+                $(#[cfg(feature = $plugin_feature:literal)])?
+                $(#[custom($plugin_meta:meta)])*
+                $($plugin_path:ident::)* : $plugin_name:ident
+            ),*
+            $(
+                ,$(
+                    #[doc(hidden)]
+                    $(#[cfg(feature = $hidden_plugin_feature:literal)])?
+                    $(#[custom($hidden_plugin_meta:meta)])*
+                    $($hidden_plugin_path:ident::)* : $hidden_plugin_name:ident
+                ),+
+            )?
+
+            $(,)?
+        }
+        $($(#[doc = $post_doc:literal])+)?
+    ) => {
+        $(#[$group_meta])*
+        ///
+        $(#[doc = concat!(
+            "* [`", stringify!($plugin_name), "`](" $(, stringify!($plugin_path), "::")*, stringify!($plugin_name), ")"
+            $(, " - with feature `", $plugin_feature, "`")?
+        )])*
+        $(
+            ///
+            $(#[doc = $post_doc])+
+        )?
+        pub struct $group;
+        impl PluginGroup for $group {
+            fn build(self) -> PluginGroupBuilder {
+                let mut group = PluginGroupBuilder::start::<Self>();
```

Comment:
This will fail to compile if `PluginGroup` and `PluginGroupBuilder` are not imported. Using `$crate` will fix this.

Additionally, `#[allow(unused_mut)]` will allow an empty `plugin_group!(MyPlugins {})` to build without warnings.

```suggestion
        impl $crate::PluginGroup for $group {
            fn build(self) -> $crate::PluginGroupBuilder {
                #[allow(unused_mut)]
                let mut group = $crate::PluginGroupBuilder::start::<Self>();
```

Reply:
Why would you want to create a plugin group with no plugins?

Reply:
I don't think there is any long-term use of having an empty plugin group, but I could see someone testing the macro empty and being confused why `rustc` is raising an warning on code they can't see. Additionally the macro doesn't prevent you from creating an empty group, so it's bound to happen at some point.

I don't think the extra `#[allow(...)]` annotation will be confusing for future reviewers, but feel free to add a comment explaining why it's there. It can't hurt :)